### PR TITLE
[WebView] [TCSACR-219] Add Policy Decision APIs

### PR DESCRIPTION
**Pull request by BasavarajPS**
[WebView] [TCSACR-219] Add Policy Decision APIs

Signed-off-by: basavarajps <basavaraj.ps@samsung.com>

### API Changes ###

 - ACR: http://suprem.sec.samsung.net/jira/browse/TCSACR-219

### CODE REVIEW

File: src/Tizen.WebView/Interop/Interop.ChromiumEwk.PolicyDecision.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            IntPtr ptr = _ewk_policy_decision_scheme_get(policydecision);
+            return Marshal.PtrToStringAnsi(ptr);
+        }
+
+        [DllImport(Libraries.ChromiumEwk, EntryPoint = "ewk_policy_decision_cookie_get", CallingConvention = CallingConvention.Cdecl, ExactSpelling = true, CharSet = CharSet.Ansi)]
+        internal static extern IntPtr _ewk_policy_decision_cookie_get(IntPtr policydecision);
+        internal static string ewk_policy_decision_cookie_get(IntPtr policydecision)
+        {
+            IntPtr ptr = _ewk_policy_decision_url_get(policydecision);
```
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Interop/Interop.ChromiumEwk.PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+namespace Tizen.WebView
+{
+    /// <summary>
+    /// Enumeration values used to specify Navigation Type.
+    /// </summary>
+    public enum NavigationType {
+        /// <summary>
+        /// Link clicked.
+        /// </summary>
+        LinkClcked = 0,
```
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+using System;
+using System.ComponentModel;
+
+namespace Tizen.WebView
+{
+    /// <summary>
+    /// Enumeration values used to specify Navigation Type.
+    /// </summary>
+    public enum NavigationType {
```
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        /// <summary>
+        /// Other.
+        /// </summary>
+        Other = 5
+    };
+
+    /// <summary>
+    /// Enumeration values used to specify Decision Type.
+    /// </summary>
+    public enum DecisionType {
```
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        public bool IsMainFrame()
+        {
+            return Interop.ChromiumEwk.ewk_policy_decision_is_main_frame(_handle);
+        }
+
+        /// <summary>
+        /// Ignores the action which triggers this decision.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public bool Ignore()
```
Comment (m-macias) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: It should be void, in C# we don't return boolean on success.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+        {
+            return Interop.ChromiumEwk.ewk_policy_decision_ignore(_handle);
+        }
+
+        /// <summary>
+        /// Accepts the action which triggers this decision.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public bool Use()
```
Comment (m-macias) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: It should be void, in C# we don't return boolean on success. I suggest writing a code example for these methods.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs (hunk 5 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs (hunk 6 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs [hunk 7]
```diff
@@ -0,0 +1,188 @@
+﻿/*
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: File name should be PolicyDecisionEventArgs.cs.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+        /// Ignore.
+        /// </summary>
+        Ignore
+    };
+
+    /// <summary>
+    /// This class provides the properties for Policy Decision of WebView.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class PolicyDecisionArgs : EventArgs
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: How about PolicyDecisionEventArgs?

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+        /// </summary>
+        Download,
+        /// <summary>
+        /// Ignore.
+        /// </summary>
+        Ignore
+    };
+
+    /// <summary>
+    /// This class provides the properties for Policy Decision of WebView.
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Plz update summary properly.

File: src/Tizen.WebView/Tizen.WebView/WebView.cs [hunk 10]
```diff
@@ -77,6 +80,24 @@ public class WebView: EvasObject
         /// <since_tizen> 4 </since_tizen>
         public event EventHandler<SmartCallbackArgs> UrlChanged;
 
+        /// <summary>
+        /// Event that occurs when the load throws an error.
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/WebView.cs]: Plz update summary properly.

File: src/Tizen.WebView/Tizen.WebView/WebView.cs [hunk 11]
```diff
@@ -77,6 +80,24 @@ public class WebView: EvasObject
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<PolicyDecisionArgs> PolicyNavigationDecide;
+
+        /// <summary>
+        /// Event that occurs when the title of the main frame is changed.
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/WebView.cs]: Plz update summary properly.

File: src/Tizen.WebView/Tizen.WebView/WebView.cs [hunk 12]
```diff
@@ -77,6 +80,24 @@ public class WebView: EvasObject
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<PolicyDecisionArgs> PolicyNewWindowDecide;
+
+        /// <summary>
+        /// Event that occurs when the URL of the main frame is changed.
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/WebView.cs]: Plz update summary properly.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs (hunk 8 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs (hunk 9 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/WebView.cs (hunk 10 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/WebView.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs (hunk 7 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/WebView.cs (hunk 11 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/WebView.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/WebView.cs (hunk 12 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/WebView.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+
+using System;
+using System.ComponentModel;
+
+namespace Tizen.WebView
+{
+    /// <summary>
+    /// Enumeration values used to specify Policy Navigation Type.
+    /// </summary>
+    public enum PolicyNavigationType {
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: I think, `NavigationPolicy` is more readable

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+        /// <summary>
+        /// Other.
+        /// </summary>
+        Other = 5
+    };
+
+    /// <summary>
+    /// Enumeration values used to specify Policy Decision Type.
+    /// </summary>
+    public enum PolicyDecisionType {
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: I think, `DecisionPolicy` is more readable

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 15]
```diff
@@ -0,0 +1,189 @@
...
+        /// <summary>
+        /// Use.
+        /// </summary>
+        Use,
+        /// <summary>
+        /// Download.
+        /// </summary>
+        Download,
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: What is means of Download?

File: src/Tizen.WebView/Tizen.WebView/WebView.cs [hunk 16]
```diff
@@ -77,6 +80,24 @@ public class WebView: EvasObject
         /// <since_tizen> 4 </since_tizen>
         public event EventHandler<SmartCallbackArgs> UrlChanged;
 
+        /// <summary>
+        /// Event that occurs when the policy navigation is decided.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<PolicyDecisionEventArgs> PolicyNavigationDecide;
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/WebView.cs]:
```suggestion
        public event EventHandler<PolicyDecisionEventArgs> NavigationPolicyDecideRequested;
```

File: src/Tizen.WebView/Tizen.WebView/WebView.cs [hunk 17]
```diff
@@ -77,6 +80,24 @@ public class WebView: EvasObject
...
+
+        /// <summary>
+        /// Event that occurs when the policy new window is decided.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<PolicyDecisionEventArgs> PolicyNewWindowDecide;
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/WebView.cs]:
```suggestion
        public event EventHandler<PolicyDecisionEventArgs> NewWindowPolicyDecideRequested;
```

File: src/Tizen.WebView/Tizen.WebView/WebView.cs [hunk 18]
```diff
@@ -77,6 +80,24 @@ public class WebView: EvasObject
...
+
+        /// <summary>
+        /// Event that occurs when the policy response is decided.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<PolicyDecisionEventArgs> PolicyResponseDecide;
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/WebView.cs]:
```suggestion
        public event EventHandler<PolicyDecisionEventArgs> ResponsePolicyDecideRequested;
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 14 above)
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: How about `NavigationPolicy`?

File: src/Tizen.WebView/Tizen.WebView/WebView.cs (hunk 16 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/WebView.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/WebView.cs (hunk 17 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/WebView.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/WebView.cs (hunk 18 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/WebView.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+        public string Scheme
+        {
+            get
+            {
+                return Interop.ChromiumEwk.ewk_policy_decision_scheme_get(_handle);
+            }
+        }
+
+        /// <summary>
+        /// Gets response status code.
```
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 14 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 13 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 15 above)
Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: I can't understand. could you explain with some example?
What is means of `Ignore`?

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 20]
```diff
@@ -0,0 +1,189 @@
...
+        /// </summary>
+        Download,
+        /// <summary>
+        /// Ignore.
+        /// </summary>
+        Ignore
+    };
+
+    /// <summary>
+    /// Arguments from the policy decision smart event callback.
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: do not use `smart event callback`
C# does not use `smart event callback`

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 21]
```diff
@@ -0,0 +1,189 @@
...
+    /// This class also rovides the properties for Policy Decision of WebView.
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]:
```suggestion
    /// This class also provides the properties for Policy Decision of WebView.
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 22]
```diff
@@ -0,0 +1,189 @@
...
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class PolicyDecisionEventArgs : EventArgs
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: This class is used in various policy decision events.
I can't understand why do you provide `NavigationPolicy DecisionType` property?
It only valid for `NavigationPolicyDecideRequested`. 
What means of `DecisionType` in others event?

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 15 above)
Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: This enum used for all decision request event, but name is `NavigationPolicy`. 
Is it right?

Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Ignore: If user (Application) does not want to continue loading Navigation Policy is set as Ignore, eg: for custom URLs like tizen://tizenstore/app.wgt
Use: If user (Application) wants to continue loading the webpage policy is set as use.
Download: when the URL is downloadable type ex: http://abc.com/filenamae.apk, Navigation Policy Set as Download and same is informed to Application, so that app will send the URL to downloader module

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 21 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 15 above)
Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: no way to set NavigationPolicy

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 22 above)
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: So, are you suggesting that we should provide separated EventArgs for each event?

Comment (myroot) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Yes, you should separate a event args class

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 23]
```diff
@@ -0,0 +1,250 @@
...
+    /// <summary>
+    /// Arguments from the policy decision smart event callback.
+    /// This class also provides the properties for Policy Decision of WebView.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public abstract class PolicyDecisionEventArgs : EventArgs
+    {
+        private IntPtr _handle;
+
+        public PolicyDecisionEventArgs(IntPtr handle)
```
[the remark this answers is not in the document]
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]:
```suggestion
        internal PolicyDecisionEventArgs(IntPtr handle)
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 24]
```diff
@@ -0,0 +1,250 @@
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public void Use()
+        {
+            Interop.ChromiumEwk.ewk_policy_decision_use(_handle);
+        }
+
+    }
+
+    public class NewWindowPolicyEventArgs : PolicyDecisionEventArgs
```
[the remark this answers is not in the document]
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Plz add a description for this class.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 25]
```diff
@@ -0,0 +1,250 @@
...
+        {
+            Interop.ChromiumEwk.ewk_policy_decision_use(_handle);
+        }
+
+    }
+
+    public class NewWindowPolicyEventArgs : PolicyDecisionEventArgs 
+    {
+        private IntPtr _handle;
+        public NewWindowPolicyEventArgs(IntPtr handle) : base(handle)
```
[the remark this answers is not in the document]
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]:
```suggestion
        internal NewWindowPolicyEventArgs(IntPtr handle) : base(handle)
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 26]
```diff
@@ -0,0 +1,250 @@
...
+        {
+            return new NewWindowPolicyEventArgs(info);
+        }
+    }
+
+
+    public class NavigationPolicyEventArgs : PolicyDecisionEventArgs
+    {
+        private IntPtr _handle;
+        public NavigationPolicyEventArgs(IntPtr handle) : base(handle)
```
[the remark this answers is not in the document]
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]:
```suggestion
        internal NavigationPolicyEventArgs(IntPtr handle) : base(handle)
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 27]
```diff
@@ -0,0 +1,250 @@
...
+        }
+        
+        internal static NewWindowPolicyEventArgs CreateFromSmartEvent(IntPtr data, IntPtr obj, IntPtr info)
+        {
+            return new NewWindowPolicyEventArgs(info);
+        }
+    }
+
+
+    public class NavigationPolicyEventArgs : PolicyDecisionEventArgs
```
[the remark this answers is not in the document]
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Plz add a description for this class.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 28]
```diff
@@ -0,0 +1,250 @@
...
+            return Interop.ChromiumEwk.ewk_policy_decision_is_main_frame(_handle);
+        }
+        
+        internal static NavigationPolicyEventArgs CreateFromSmartEvent(IntPtr data, IntPtr obj, IntPtr info)
+        {
+            return new NavigationPolicyEventArgs(info);
+        }
+    }
+
+    public class ResponsePolicyEventArgs : PolicyDecisionEventArgs
```
[the remark this answers is not in the document]
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Plz add a description for this class.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 29]
```diff
@@ -0,0 +1,250 @@
...
+    {
+        private IntPtr _handle;
+        public ResponsePolicyEventArgs(IntPtr handle) : base(handle)
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]:
```suggestion
        internal ResponsePolicyEventArgs(IntPtr handle) : base(handle)
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 30]
```diff
@@ -0,0 +1,250 @@
...
+            {
+                return Interop.ChromiumEwk.ewk_policy_decision_cookie_get(_handle);
+            }
+        }
+        
+        /// <summary>
+        /// Checks Navigation Type is Download.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public bool isDownload
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]:
```suggestion
        public bool IsDownload
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 31]
```diff
@@ -0,0 +1,250 @@
...
+        /// <summary>
+        /// Other.
+        /// </summary>
+        Other = 5
+    };
+
+    /// <summary>
+    /// Enumeration values used to specify Navigation Policy.
+    /// </summary>
+    public enum NavigationPolicy {
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Plz move this enum to Interop.ChromiumEwk.PolicyDecision.cs

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 32]
```diff
@@ -0,0 +1,250 @@
...
+        /// </summary>
+        Download,
+        /// <summary>
+        /// Ignore.
+        /// </summary>
+        Ignore
+    };
+
+    /// <summary>
+    /// Arguments from the policy decision smart event callback.
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]:
```suggestion
    /// Arguments from the policy decision events.
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 33]
```diff
@@ -0,0 +1,250 @@
...
+                return Interop.ChromiumEwk.ewk_policy_decision_cookie_get(_handle);
+            }
+        }
+
+        /// <summary>
+        /// Checks if frame requested in policy decision is main frame.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool IsMainFrame()
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Plz move this method to `PolicyDecisionEventArgs`, it should be provided for all of policy event.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 29 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 30 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 32 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 33 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 25 above)
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 24 above)
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 23 above)
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 27 above)
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 26 above)
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 28 above)
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 29 above)
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 34]
```diff
@@ -0,0 +1,206 @@
...
+        internal NavigationPolicyEventArgs(IntPtr handle) : base(handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Checks Navigation Type Is Link Clicked.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public bool IsLinkClicked
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Plz revert this change, NavigationType should be provided.

File: src/Tizen.WebView/Interop/Interop.ChromiumEwk.PolicyDecision.cs [hunk 35]
```diff
@@ -0,0 +1,85 @@
...
+
+using System;
+using System.Runtime.InteropServices;
+using Tizen.WebView;
+
+internal static partial class Interop
+{
+    internal static partial class ChromiumEwk
+    {
+        public enum NavigationType {
```
[the remark this answers is not in the document]
Comment (BasavarajPS) [reply src/Tizen.WebView/Interop/Interop.ChromiumEwk.PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 34 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Interop/Interop.ChromiumEwk.PolicyDecision.cs [hunk 36]
```diff
@@ -0,0 +1,76 @@
...
+
+using System;
+using System.Runtime.InteropServices;
+using Tizen.WebView;
+
+internal static partial class Interop
+{
+    internal static partial class ChromiumEwk
+    {
+        public enum NavigationPolicy {
```
Comment (younghajung) [inline src/Tizen.WebView/Interop/Interop.ChromiumEwk.PolicyDecision.cs]:
```suggestion
        public enum PolicyDecisionType {
```

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 37]
```diff
@@ -0,0 +1,236 @@
...
+        public string Cookie
+        {
+            get
+            {
+                return Interop.ChromiumEwk.ewk_policy_decision_cookie_get(_handle);
+            }
+        }
+        
+        /// <summary>
+        /// Checks Navigation Type Is Download.
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]:
```suggestion
        /// Checks policy decision type is Download.
```

File: src/Tizen.WebView/Interop/Interop.ChromiumEwk.PolicyDecision.cs (hunk 36 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Interop/Interop.ChromiumEwk.PolicyDecision.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 38]
```diff
@@ -0,0 +1,236 @@
...
+        /// <summary>
+        /// Gets the Navigation Type of policy.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public NavigationType NavigationType
+        {
+            get
+            {
+                return Interop.ChromiumEwk.ewk_policy_decision_navigation_type_get(_handle);
+	    }
```
Comment (younghajung) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Plz check indentation.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 39]
```diff
@@ -0,0 +1,236 @@
...
+    public class ResponsePolicyEventArgs : PolicyDecisionEventArgs
+    {
+        private IntPtr _handle;
+        internal ResponsePolicyEventArgs(IntPtr handle) : base(handle)
+        {
+            _handle = handle;
+        }
+
+        /// <summary>
+        /// Gets response status code.
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Could explain about response status code?
Is it means a http renpose code?
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs [hunk 40]
```diff
@@ -0,0 +1,236 @@
...
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public int ResponseStatusCode
```
Comment (myroot) [inline src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: How about StatusCode?

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 39 above)
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Yes, it is.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 40 above)
Comment (younghajung) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: We will accept your suggestion.

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 37 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 38 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done

File: src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs (hunk 40 above)
Comment (BasavarajPS) [reply src/Tizen.WebView/Tizen.WebView/PolicyDecisionEventArgs.cs]: Done
